compiler/resolution/functionResolution.cpp

```diff
@@ -9175,9 +9175,9 @@ static void moveHaltMoveIsUnacceptable(CallExpr* call) {
   } else {
     if (lhsSym->hasFlag(FLAG_TYPE_VARIABLE) == false &&
         lhsSym->hasFlag(FLAG_MAYBE_TYPE)    == false) {
-      FnSymbol* fn = toFnSymbol(call->parentSymbol);
 
-      if (fn->getReturnSymbol() == lhsSym) {
+      if (auto fn = toFnSymbol(call->parentSymbol);
```

Comment:
Since when have we been able to put semicolons in conditionals in C++ ?!?

Anyway I think it's more readable as

```
auto fn = toFnSymbol(call->parentSymbol);
if (fn && fn->getReturnSymbol() == lhsSym) {
```

Reply:
Its a C++17 feature, if statement with an initializer. It just makes the scope of `fn` smaller. But I will rewrite the other way.